### ISSUE TITLE
Allow to document events that are not explicitly in the component file

### ISSUE THREAD

**Issue by elevatebart** [vue-styleguidist/vue-styleguidist#836]
**The problem**

Sometimes, events are triggered in functions or in other items that the library does not parse.
It would be useful to document those nonetheless.  

**Proposed solution**

I was thinking something like this as an optional additional block. 
Still need to figure out how to clarify that both

```js
/** 
   * Define the event just before the function block
   *
   * @event updating
   * @property { String } - the first prop given by the event
   */
 /** 
   * Some comment for the function
   * @fires updating
   * @arg { String } name updated property name
   * @arg newValue new value that we want to update
   */
  onUpdate (name, newValue) {
    // some external method, for example from a method
    // (bind to the Vue instance) provided by a standard js class
  }
 
```

`@fires` would look for the event definition in the comments besides the function and use it to show users.

**Alternative solutions**

Parse functions called by each function to detect event calling there... meh

**Additional context**

closes https://github.com/vue-styleguidist/vue-styleguidist/issues/586
